Sources/TuistKit/Utils/ImportSourceCodeScanner.swift

```diff
@@ -5,65 +5,84 @@ enum ProgrammingLanguage {
     case objc
 }
 
+struct Match {
+    let module: String
+    let range: Range<String.Index>
+}
+
 final class ImportSourceCodeScanner {
     func extractImports(from sourceCode: String, language: ProgrammingLanguage) throws -> [String] {
         switch language {
         case .swift:
-            try extractAllImportsSwift(from: sourceCode)
+            try extract(from: sourceCode, language: .swift)
         case .objc:
-            try extractAllImportsObjc(from: sourceCode)
+            try extract(from: sourceCode, language: .objc)
         }
     }
 
-    private func extractAllImportsSwift(from code: String) throws -> [String] {
-        let pattern = #"import\s+(?:struct\s+|enum\s+|class\s+)?([\w]+)"#
+    private func extract(from code: String, language: ProgrammingLanguage) throws -> [String] {
+        let pattern = switch language {
+        case .swift:
+            #"import\s+(?:struct\s+|enum\s+|class\s+)?([\w]+)"#
+        case .objc:
+            "@import\\s+([A-Za-z_0-9]+)|#(?:import|include)\\s+<([A-Za-z_0-9-]+)/"
+        }
 
-        var modules: [String] = []
+        let codeWithoutComments = try CommentsRemover.removeComments(from: code)
+
+        var result: [String] = []
 
         let regex = try NSRegularExpression(pattern: pattern, options: [])
 
-        let matches = regex.matches(
-            in: code,
-            options: [],
-            range: NSRange(
-                location: 0,
-                length: code.utf16.count
-            )
-        )
+        let lines = codeWithoutComments.components(separatedBy: .newlines)
 
-        for match in matches {
-            if let moduleRange = Range(match.range(at: 1), in: code) {
-                let module = String(code[moduleRange])
-                let topLevelModule = module.split(separator: ".").first.map(String.init)
-                if let topLevelModule, !modules.contains(topLevelModule) {
-                    modules.append(topLevelModule)
+        for line in lines {
+            let range = NSRange(location: 0, length: line.utf16.count)
+            let matches = regex.matches(in: line, options: [], range: range)
+
+            for match in matches {
+                let match = switch language {
+                case .swift:
+                    processMatchSwift(match: match, line: line)
+                case .objc:
+                    processMatchObjc(match: match, line: line)
+                }
+                if let match {
+                    if !result.contains(where: { $0 == match.module }) {
+                        result.append(match.module)
+                    }
                 }
             }
```

Comment:
This whole for loop can be turned into a `compactMap` and we can use `Set` to filter out duplicates:
```swift
return Set(
   codeWithoutComments
    .components(separatedBy: .newlines)
   .compactMap { line in
      ...
   }
)
```